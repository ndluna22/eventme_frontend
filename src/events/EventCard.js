import React from "react";
import { Link } from "react-router-dom";
import "./EventCard.css";
import AspectRatio from "@mui/joy/AspectRatio";
import Button from "@mui/joy/Button";
import Card from "@mui/joy/Card";
import CardContent from "@mui/joy/CardContent";
import Typography from "@mui/joy/Typography";

/** Show limited information about an event
 *
 * Is rendered by EventList to show a "card" for each event.
 *
 * EventList -> EventCard
 */

function EventCard({ id, name, images, category, startDate, city, state }) {
  console.debug("EventCard", name);

  return (
    <Card sx={{ width: 300 }}>
      <div>
        <Typography level="title-lg">{name}</Typography>
        <Typography level="body-sm"></Typography>
        {/* <IconButton
  aria-label="bookmark
  variant="plain"
  color="neutral"
  size="sm"
  sx={{ position: "absolute", top: "0.875rem", right: "0.5rem" }}
>
  <FavoriteAdd />
</IconButton> */}
      </div>
      <AspectRatio minHeight="120px" maxHeight="200px">
        <img src={images} srcSet={images} loading="lazy" alt="" />
      </AspectRatio>
      <CardContent orientation="horizontal">
        <div>
          <Typography fontSize="lg" fontWeight="lg">
            {category}
          </Typography>
          <Typography level="body-sm">Date: {startDate}</Typography>
          <Typography level="body-sm">
            {city}, {state}
          </Typography>
        </div>
        <Link to={`/events/${id}`}>
          <Button
            variant="solid"
            size="md"
            color="primary"
            sx={{ ml: "auto", alignSelf: "center", fontWeight: 600 }}
          >
            Explore
          </Button>
        </Link>
      </CardContent>
    </Card>
  );
}

export default EventCard;
