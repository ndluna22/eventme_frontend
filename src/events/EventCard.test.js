import React from "react";
import { render } from "@testing-library/react";
import EventCard from "./EventCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <EventCard
        id="Z7r9jZ1AdJ4aP"
        name="Utah Jazz vs. Phoenix Suns"
        images="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <EventCard
        id="Z7r9jZ1AdJ4aP"
        name="Utah Jazz vs. Phoenix Suns"
        images="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
