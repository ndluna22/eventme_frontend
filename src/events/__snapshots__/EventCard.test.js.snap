// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-ij021v-JoyCard-root"
  >
    <div
      data-first-child=""
    >
      <p
        class="MuiTypography-root MuiTypography-title-lg css-ar9ysw-JoyTypography-root"
      >
        Utah Jazz vs. Phoenix Suns
      </p>
      <p
        class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
      />
    </div>
    <div
      class="MuiAspectRatio-root css-lqq0j9-JoyAspectRatio-root"
    >
      <div
        class="MuiAspectRatio-content MuiAspectRatio-variantSoft MuiAspectRatio-colorNeutral css-6e7lq1-JoyAspectRatio-content"
      >
        <img
          alt=""
          data-first-child=""
          loading="lazy"
          src="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
          srcset="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </div>
    </div>
    <div
      class="MuiCardContent-root css-vpmu79-JoyCardContent-root"
      data-last-child=""
    >
      <div>
        <p
          class="MuiTypography-root MuiTypography-body-md css-1ygy07m-JoyTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
        >
          Date: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
        >
          , 
        </p>
      </div>
      <a
        href="/events/Z7r9jZ1AdJ4aP"
      >
        <button
          class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-cb19rj-JoyButton-root"
          type="button"
        >
          Explore
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <div
    class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-ij021v-JoyCard-root"
  >
    <div
      data-first-child=""
    >
      <p
        class="MuiTypography-root MuiTypography-title-lg css-ar9ysw-JoyTypography-root"
      >
        Utah Jazz vs. Phoenix Suns
      </p>
      <p
        class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
      />
    </div>
    <div
      class="MuiAspectRatio-root css-lqq0j9-JoyAspectRatio-root"
    >
      <div
        class="MuiAspectRatio-content MuiAspectRatio-variantSoft MuiAspectRatio-colorNeutral css-6e7lq1-JoyAspectRatio-content"
      >
        <img
          alt=""
          data-first-child=""
          loading="lazy"
          src="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
          srcset="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </div>
    </div>
    <div
      class="MuiCardContent-root css-vpmu79-JoyCardContent-root"
      data-last-child=""
    >
      <div>
        <p
          class="MuiTypography-root MuiTypography-body-md css-1ygy07m-JoyTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
        >
          Date: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
        >
          , 
        </p>
      </div>
      <a
        href="/events/Z7r9jZ1AdJ4aP"
      >
        <button
          class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-cb19rj-JoyButton-root"
          type="button"
        >
          Explore
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
